### PR TITLE
Feat/deliverable1

### PR DESCRIPTION
**Pull request by DavidTG7**
This is the Pull Request for deliverable1.

Created:
components dir :
Header dir: Header.jsx, Header.styled.js, index.js, Header.test.jsx
Home dir:
Slogan dir: Slogan.jsx, Slogan.styled.js, index.js
Slider dir: Slider.jsx, Slider.styled.js, index. js
ProductCategories dir: ProductCategories.jsx, Productcategories.styled.js, index.js
FeaturedProducts dir: FeaturedProducts.jsx, FeaturedProducts.styled.js, index,js
Home.jsx, Home.styled.js, index.js, Home.test.jsx
Footer dir: Footer.jsx, Footer.styled.js, index.js
index.js
Images dir.
utils dir: constant.styles.js
styles.css

Updated:
App.jsx, setupTests.js

Link Deployment: https://davidtg7.github.io/2022React-Bootcamp-CapstoneProject/

Thank you.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useFeaturedBanners } from './utils/hooks/useFeaturedBanners';
+import { Header, Home, Footer } from './components';
+
+function App() {
+  const { data, isLoading } = useFeaturedBanners();
+  console.log(data, isLoading);
```
Comment (franciscogranadoswizeline) [inline src/App.jsx:7]: if the console log is not required remove it

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  return (
+    <>
+      <Header />
+      <Home />
+      <Footer text='Ecommerce created during Wizeline’s Academy React Bootcamp' />
```
Comment (franciscogranadoswizeline) [inline src/App.jsx:13]: in this case, the footer is going to be re-used in another section? if the awnser is yer...ij ok to use the text as a prop....if not, maybe the text should be part of the foot state.

File: src/components/Footer/Footer.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <FooterMain>
+      {text}
+    </FooterMain>
+  );
+}
+
+export default Footer;
+
+// Footer.defaultProps = {
```
Comment (franciscogranadoswizeline) [inline src/components/Footer/Footer.jsx]: remove comments

File: src/components/Footer/FooterMain.styled.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+const FooterMain = styled.footer`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 2.5rem;
+  background: ${yellowMain};
+  font-size: .8rem;
+  color: ${bluePrince};
```
Comment (franciscogranadoswizeline) [inline src/components/Footer/FooterMain.styled.js:12]: i like this, good jog. re-use all colors constants is really goog.

File: src/components/Footer/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import Footer from "./Footer";
+
+export default Footer;
```
Comment (franciscogranadoswizeline) [inline src/components/Footer/index.js:3]: why is imported also from here the footer? this approach is good but it's benefits are when you define an index by module.

File: src/components/Header/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+} from './Header.styled';
+
+function Header() {
+  return (
+    <HeaderMain isYellow>
+      <Image src={logo} alt="logo" />
+      <Name>
+        Furny
+      </Name>
+      <Search type="text" placeholder="Search" />
```
Comment (franciscogranadoswizeline) [inline src/components/Header/Header.jsx]: InputSearch ....what do you think? i consider that InputSearch add more semantic

File: src/components/Home/FeaturedProducts/FeaturedProducts.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { ProductImage, ProductGrid, ImageWrap, InfoImage } from './FeaturedProducts.styled';
+import FeaturedItems from '../../../mocks/en-us/featured-products.json';
+
+const allItems = FeaturedItems.results.map((item,i) => {
```
Comment (franciscogranadoswizeline) [inline src/components/Home/FeaturedProducts/FeaturedProducts.jsx:5]: good approach

File: src/components/Home/FeaturedProducts/FeaturedProducts.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <ImageWrap key={`photo-${i}`} >
+      <ProductImage src={item.data.mainimage.url} alt='' />
```
Comment (franciscogranadoswizeline) [inline src/components/Home/FeaturedProducts/FeaturedProducts.jsx]: always the code should have an alt value, accesibility matters

File: src/components/Home/Slider/Slider.jsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
+import React, { useRef } from 'react';
+import Banners from '../../../mocks/en-us/featured-banners.json';
+import { ReactComponent as LeftArrowIcon } from '../../../images/iconmonstr-angel-left-thin.svg';
+import { ReactComponent as RightArrowIcon } from '../../../images/iconmonstr-angel-right-thin.svg';
+import { MainBox, SlideshowBox, Slide, Controls, Button } from './Slider.styled';
+
+function Slider() {
+  const slideshow = useRef(null);
+
+  const nextPic = () => {
```
Comment (franciscogranadoswizeline) [inline src/components/Home/Slider/Slider.jsx:10]: advance implementation, really good!

File: src/components/index.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import Header from './Header';
+// import Slider from './Slider';
+// import Slogan from './Slogan';
+import Footer from './Footer';
+// import ProductCategories from './ProductCategories';
```
Comment (franciscogranadoswizeline) [inline src/components/index.js]: remove comments

File: package.json [hunk 11]
```diff
@@ -77,5 +80,16 @@
       "last 1 safari version"
     ]
   },
-  "devDependencies": {}
+  "devDependencies": {
+    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.7",
+    "enzyme": "^3.11.0",
+    "eslint": "^7.32.0",
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-plugin-import": "^2.26.0",
```
Comment (franciscogranadoswizeline) [inline package.json]: are you goint to configure some pre-commit?

**Review by franciscogranadoswizeline — CHANGES_REQUESTED** [review]
Good job david, i like all the implemenations...just leave some comments about minor details. Please, before change it....let's discuss trade off

File: package.json (hunk 11 above)
Comment (DavidTG7) [reply package.json]: I juts wanted to try the Linter to improve my code, but at the end it was hard to use, so I just took it out.

File: src/components/Header/Header.jsx (hunk 6 above)
Comment (DavidTG7) [reply src/components/Header/Header.jsx]: I agree.

**Pull request merged**